Assets/Scripts/GameScene/BoardGenerator.cs

```diff
@@ -1,38 +1,50 @@
-using System.Collections;
-using System.Collections.Generic;
+using System;
 using UnityEngine;
 
 public class BoardGenerator : MonoBehaviour
 {
-    [SerializeField] private float width;
-    [SerializeField] private float height;
-    [SerializeField] private FloorTile tile;
-    
-    
-    // Start is called before the first frame update
-    void Start()
+    public static event Action<BoardGenerator> BoardDidLoad;
+    [SerializeField] private Board board; 
+    public Vector3[,] boardData { get; private set; }
+
+    private void Awake()
     {
-        GenerateBoard();
+        GenerateBoardData();
+        InstantiateBoard(boardData);
+        BoardDidLoad?.Invoke(this);
```

Comment:
is this event realted t `GenerateBoardData()` or `InstantiateBoard(boardData)`? if it does, it's a bad place to to invoke it - what happens if somebody moves one of these function out of here?! The entire process will break
Invocation should either be inside one of these functions, or pack them into another function which calls them and includes the invocation

Reply:
entered  BoardDidLoad?.Invoke(this); to InstantiateBoard. 
entered  InstantiateBoard to GenerateBoardData(changed the name to  GenerateBoard).